Library/Homebrew/os/linux/elf.rb

```diff
@@ -127,7 +130,20 @@ def initialize(path)
     private
 
     def needed_libraries(path)
-      if DevelopmentTools.locate "readelf"
+      if HOMEBREW_PATCHELF_RB
+        patcher = path.patcher
+        soname = begin
+            patcher.soname
+          rescue PatchELF::PatchError
+            nil
+          end
+        needed = begin
+            patcher.needed
+          rescue PatchELF::PatchError
```

Comment:
Ditto.

Reply:
Simple, monkey see monkey do. 
I saw the existing implementation didn't have warnings, copied that :sweat_smile:

Reply:
The existing implementation calls executables with `popen_read`, which prints `stderr` to the console when `HOMEBREW_STDERR=1` is defined.